File: 백준/Silver/11047. 동전 0/동전 0.py
n, k = map(int, input().split())
coins = [int(input()) for _ in range(n)]
coins.reverse()
result = 0

for i in range(n):
    if coins[i] > k: continue
    result += k // coins[i]
    k %= coins[i]
    
print(result)

File: 백준/Silver/11047. 동전 0/동전 0.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int N = Integer.parseInt(st.nextToken());
		int K = Integer.parseInt(st.nextToken());
		int[] coins = new int[N];
		for (int i = 0; i < N; i++) {
			coins[i] = Integer.parseInt(br.readLine());
		}
		
		int result = 0;
		for (int i = N - 1; i >= 0; i--) {
			if (coins[i] > K) continue;
			result += (K / coins[i]);
			K %= coins[i];
		}
		System.out.println(result);
	}
}